src/webui/webui.c

```diff
@@ -410,11 +428,19 @@ http_stream_run(http_connection_t *hc, profile_chain_t *prch,
       break;
 
     case SMT_GRACE:
-      grace = sm->sm_code < 5 ? 5 : grace;
+      if (sm->sm_code < 5) {
```

Comment:
In my testing `SMT_GRACE` is called immediately when starting the stream. The old version of the code would instantly downgrade the grace period from 20 seconds to 5.

I believe this was not intentional and it is a bug. I've changed it so it can only increase the period, but not decrease it.